### PR TITLE
DEPR: use_inf_as_na

### LINKED ISSUE #51684: DEPR: use_inf_as_na
Support for use_inf_as_na isn't great, it performs poorly, and we don't have any way to make EAs respect it.  Is this worth keeping?

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #51684 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by mroeschke** [comment]
```
2023-06-01T20:16:55.9250397Z docstring of pandas.reset_option:305: WARNING: Explicit markup ends without a blank line; unexpected unindent.
2023-06-01T20:16:55.9251576Z docstring of pandas.set_option:310: WARNING: Explicit markup ends without a blank line; unexpected unindent.
2023-06-01T20:16:55.9254260Z docstring of pandas.describe_option:310: WARNING: Explicit markup ends without a blank line; unexpected unindent.
2023-06-01T20:16:55.9255238Z docstring of pandas.get_option:307: WARNING: Explicit markup ends without a blank line; unexpected unindent.
```

**Comment by jbrockmendel** [comment]
It isn't clear to me what those docbuild complaints are pointing to.

File: pandas/core/config_init.py [hunk 1]
```diff
@@ -411,6 +411,8 @@ def is_terminal() -> bool:
     True means treat None, NaN, INF, -INF as NA (old way),
     False means None and NaN are null, but INF, -INF are not NA
     (new way).
+
+.. deprecated:: 2.1.0
```
Comment (mroeschke) [inline pandas/core/config_init.py:415]: I think this is the line that the doc build is complaining about. Maybe put it as the first line?

Comment (jbrockmendel) [reply pandas/core/config_init.py]: hmm that didnt seem to help

Comment (jbrockmendel) [reply pandas/core/config_init.py]: think this is handled now

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**